pallets/subtensor/src/tests/children.rs

```diff
@@ -3437,7 +3438,8 @@ fn test_childkey_take_drain() {
 #[test]
 fn test_childkey_take_drain_validator_take() {
```

Comment:
This test should be refactored into a helper function that would then be used in two tests:
- test_childkey_take_same_coldkey
- test_childkey_take_drain_validator_take

The helper function in question wouild accept parameters for `coldkey_parent` and `coldkey_child`, which would be the same in the first test and different in the other test. The other parameters of the helper would be:
- `expected_child_emission`
- `expected_parent_emission`
- `expected_nominator_emission`

This should save ~120 LOC and it will make maintenance of the test suite less costly as adjustments to the internal subtensor interface woudln't have to be replicated in two places, but only in one.